executor/joiner.go

```diff
@@ -236,16 +272,46 @@ func (j *antiSemiJoiner) tryToMatch(outer chunk.Row, inners chunk.Iterator, chk
 	return false, nil
 }
 
-func (j *antiSemiJoiner) onMissMatch(outer chunk.Row, chk *chunk.Chunk) {
+func (j *antiSemiJoiner) onMissMatchInners(outer chunk.Row, chk *chunk.Chunk) {
+	chk.AppendRow(outer)
+}
+
+// tryToMatchOuters implements joiner interface.
+func (j *antiSemiJoiner) tryToMatchOuters(outer chunk.Row, inners chunk.Iterator, chk *chunk.Chunk) (matched bool, err error) {
+	if inners.Len() == 0 {
+		return false, nil
+	}
+
+	if len(j.conditions) == 0 {
+		inners.ReachEnd()
+		return true, nil
+	}
+
+	for inner := inners.Current(); inner != inners.End(); inner = inners.Next() {
+		j.makeShallowJoinRow(j.outerIsRight, outer, inner)
+
+		matched, err = expression.EvalBool(j.ctx, j.conditions, j.shallowRow.ToRow())
+		if err != nil {
+			return false, errors.Trace(err)
+		}
+		if matched {
+			inners.ReachEnd()
+			return true, nil
+		}
+	}
+	return false, nil
+}
+
+func (j *antiSemiJoiner) onMissMatchOuters(outer chunk.Row, chk *chunk.Chunk) {
```

Comment:
@yu34po Seems all `onMissMatchInners()` is the same with `onMissMatchOuters()`?

Reply:
From the function signature and the implementation, they are the same:
```go
onMissMatchInners(outer chunk.Row, chk *chunk.Chunk)
onMissMatchOuters(outer chunk.Row, chk *chunk.Chunk)
```

From the use scenario, it's only meaningful to call `onMissMatch()` for outer rows?

Reply:
merge inner and outer onmissmatch to one